### DISCUSSION TITLE
Animation management issue in <model-viewer>

### DISCUSSION THREAD

**Discussion by stefano-ciotola** [google/model-viewer#4456] [Q&A]
### Problem Description

I've been using <model-viewer> to integrate various 3D animations into my website. I've noticed that when I pause one animation and then start a new animation afterward, the first animation resets to its initial frame instead of staying where it was paused.

### Details

Let's say we have two animations named "AnimationA" and "AnimationB". 
If "AnimationA" is paused and subsequently I try to play "AnimationB", "AnimationA" returns to its initial frame instead of staying where it was paused.

Ideally, I'd expect "AnimationA" to remain where it left off, without interfering with "AnimationB".

### Steps to Reproduce

    Load a GLTF model with at least two separate animations.
    Play "AnimationA".
    Pause "AnimationA" at any given point.
    Start "AnimationB".
    Observe that "AnimationA" resets to its initial frame.

A potential solution might be to save the current state of the animation when paused, and ensure this state isn't altered until it's played back or restarted.

I'd love to get feedback on this behavior and find out if there's a built-in function or solution in <model-viewer> that might address this issue. 

Thank you!

**Comment by elalish** [comment]
No, that's intended behavior, since our animation system is quite simple (only one at a time). All we do is fade out the old animation and fade in the new one. We don't expose a full animation mixer.

**Comment by stefano-ciotola** [comment]
Ok, thank you for your reply.

I've found an alternative solution: I'm now handling the animations by directly accessing the scene and using `mixer = new THREE.AnimationMixer(scene);`.

However, I've run into another issue regarding annotations. When I was using the integrated animations in model-viewer, the position property of the hotspot was updating automatically (I had set data-surface so it followed the animated mesh). But now, with the animation managed through ThreeJS, the hotspot no longer updates its position.

Is there a way to force such an update, or do you have any suggestions on how I might solve this issue?

Thank you

**Comment by stefano-ciotola** [comment]
I've figured out a solution myself: I simply stored all the hotspot instances (well, I'm traversing through the scene and assuming that objects with the property `isCSS2DObject === true` are hotspots, but I'm not sure if it makes sense) in an array and then invoked `updateSurface(true)` within the `animate()` function used to render the animation via ThreeJS.

If anyone has any other alternative solution, I'm open to suggestions!
